### ISSUE TITLE
Unexpected type narrowing for enums inside labeled loop statement

### ISSUE THREAD

**Issue by xuld** [microsoft/TypeScript#35323]
**TypeScript Version:**  3.8.0-dev.20191123
**Search Terms:** enum type 
**Code**
```ts
enum User { xld, zxl }

let user: User = User.xld
label: while (true) {
    switch (user) {
        case User.xld:
            user = User.zxl
            continue label
        case User.zxl: // Unexpected error: Type 'User.zxl' is not comparable to type 'User.xld'.
            break label
    }
}
```

**Expected behavior:**
No errors reported.

**Actual behavior:**
Unexpected error: Type 'User.zxl' is not comparable to type 'User.xld'.

**Playground Link:** http://www.typescriptlang.org/play/index.html?ssl=12&ssc=2&pln=1&pc=1#code/KYOwrgtgBAqgzsATlA3lAHgGwCYBooBeWUAvgLABQlmwALlGAogFyxNQC8bSAdFttQCGAI2CZWAdwAWASxpQAFLURhgASlSUo2qHAkzaAYymLGSDSi07rhwQm6I+OZletuG7LvF5FMr9zqGAPYgtDLgwFCYImL+NnaR3o6+LhQBbsKIwIIA1lExfmk65BQkQA

**Related Issues:** None
The value of variable `user` is updated to `User.zxl` inside loop. 
If I omit label and just use `continue`, it works correctly.
